components/collective-page/hero/Hero.js

```diff
@@ -242,28 +244,22 @@ const Hero = ({ collective, host, isAdmin, onPrimaryColorChange }) => {
                       values={{
                         FiscalHost: <DefinedTerm term={Terms.FISCAL_HOST} color="black.700" />,
                         hostName: (
-                          <LinkCollective collective={host}>
-                            <Span data-cy="fiscalHostName" color="black.700">
-                              {host.name}
-                            </Span>
-                          </LinkCollective>
+                          <OverflowCollectiveName
+                            collective={host}
+                            dataCy="fiscalHostName"
+                            cutOff={parentIsHost && host.name.length >= 30 ? host.name.length : 30}
+                          />
                         ),
                       }}
                     />
                   </Container>
-                  {collective.connectedTo.length !== 0 && (
+                  {!parentIsHost && (
```

Comment:
@simonv3 : I think this will error out the page when you got to a collective page. When `collective.connectedTo.length = 0`, the condition `!parentIsHost` evaluates to `false` and then then there will be a problem at ` <OverflowCollectiveName collective={collective.connectedTo[0].collective} />` since `collective.connectedTo[0]` will be undefined. 🤔